Include/pythonrun.h

```diff
@@ -25,8 +25,13 @@ PyAPI_DATA(int) (*PyOS_InputHook)(void);
  * no two calls to check recursion depth are more than this far
  * apart. In practice, that means it must be larger than the C
  * stack consumption of PyEval_EvalDefault */
-#if defined(Py_DEBUG) && defined(WIN32)
+#if defined(_Py_ADDRESS_SANITIZER) || defined(_Py_THREAD_SANITIZER)
+#  define PYOS_STACK_MARGIN 4096
+#elif defined(Py_DEBUG) && defined(WIN32)
 #  define PYOS_STACK_MARGIN 3072
+#elif defined(__wasi__)
+   /* Web assembly has two stacks, so this isn't really a size */
```

Comment:
Mind clarifying what it is, if not a size?

Reply:
Currently, it's a hacky guess at the amount of C stack used before the webassembly stack overflows. 
Which is why https://github.com/python/cpython/issues/130397 exists.